src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1403,15 +1396,14 @@ private async ValueTask<(Socket?, Stream, TransportContext?)> ConnectAsync(HttpR
                 stream = sslStream;
             }
 
-            return (socket, stream, transportContext);
+            return (stream, transportContext);
```

Comment:
If I get this right, before this change, in case of `SslStream` we would still have access to the socket, but now we don't. What implications does this have? We would in all https sync cases issue an async read instead of a socket poll, am I right? Is this a trade off we're fine with? If so, does is make sense to still keep the socket poll just for sync, plain text cases?

cc @stephentoub

Reply:
I think there are two main concerns.

First, when an https connection is put back into the pool and we first go to check it to determine whether it's been closed as part of scavenging, we will now end up issuing an async read on the stream rather than polling the underlying socket.  The impact of that previously was that we'd cause the SslStream to allocate _and on Windows pin_ a buffer, which we've seen to have impactful scalability impact.  That's what https://github.com/dotnet/runtime/pull/49474 fixed, ensuring we had this socket available even for SslStream.  However, since then we changed how we do the polling to use a zero-byte read and we taught SslStream how to avoid the allocation and pinning in the case of a zero-byte read, so those previous concerns _should_ no longer be an issue.  I think this should be fine, but we should double-check.

Second, there's also a concern about HTTP/1.1 sync operations.  There are two interesting cases here: an https connection that's been in the pool for a while, and an https connection that's relatively new to the pool.  In the former case, our scavenging logic will have already caused an async read to be performed on the stream.  In that case, when we go to reuse the connection, we're already forced to perform a sync-over-async blocking operation in order to consume that initial read if it hasn't yet completed, so for this case, changing PrepareForReuse to not look for a socket isn't impacting anything.  However, for an https connection that's relatively new to the pool and thus hasn't been checked during scavenging, it won't yet have an async read pending, and in that case, we would have used socket.Poll (assuming this was a normal https connection not through a proxy) rather than an async read ahead, so we'll potentially be forcing the request down the sync-over-async path.

I don't think we should maintain both approaches.  I think we should either stick with what we currently have (i.e. close this PR) or we should go ahead with the PR as-is and accept that we'll be potentially introducing some more sync-over-async when the right circumstances occur: HTTP/1.1, https, non-proxied, sync operation, relatively new connection to the pool, relatively slow response so that the initial read isn't complete by the time we go to consume it and end up needing to block waiting for it.  It might be worth some stress testing to see what impact this has, but I suspect it's relatively minor, all things considered.

Reply:
> I think we should either stick with what we currently have (i.e. close this PR) or we should go ahead with the PR as-is and accept that we'll be potentially introducing some more sync-over-async

I agree, and I lean towards taking this PR, primarily because it helps with https://github.com/dotnet/runtime/issues/66223. That is, if we are going to make the interaction between the connection and the pool happen through public APIs, then it would be nice to simplify those APIs by avoiding the Socket handle here.